### ISSUE TITLE
Prometheus operator does not deploy on minikube

### ISSUE THREAD

**Issue by ngehani** [coreos/kube-prometheus#426]
**What did you do?**
I ran the commands as per the README for the kube-prometheus installation

**Did you expect to see some different?**
I had it running before and now I can't seem to get it working
```
kubectl get all --all-namespaces
NAMESPACE     NAME                                       READY   STATUS             RESTARTS   AGE
kube-system   pod/coredns-5644d7b6d9-hd2g4               1/1     Running            0          6m50s
kube-system   pod/coredns-5644d7b6d9-tvwt8               1/1     Running            0          6m50s
kube-system   pod/etcd-minikube                          1/1     Running            0          6m1s
kube-system   pod/kube-apiserver-minikube                1/1     Running            0          5m43s
kube-system   pod/kube-controller-manager-minikube       1/1     Running            0          5m58s
kube-system   pod/kube-proxy-7v5rm                       1/1     Running            0          6m49s
kube-system   pod/kube-scheduler-minikube                1/1     Running            0          5m45s
kube-system   pod/storage-provisioner                    1/1     Running            1          6m55s
monitoring    pod/grafana-76b8d59b9b-vglpt               1/1     Running            0          4m10s
monitoring    pod/kube-state-metrics-5859ffdc64-n8tmf    1/1     Running            0          4m10s
monitoring    pod/node-exporter-q9k2c                    2/2     Running            0          4m10s
monitoring    pod/prometheus-adapter-5cd5798d96-x98nb    1/1     Running            0          4m9s
monitoring    pod/prometheus-operator-5d94cdc9bf-jzh2c   0/1     CrashLoopBackOff   5          6m2s

NAMESPACE     NAME                          TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                  AGE
default       service/kubernetes            ClusterIP   10.96.0.1        <none>        443/TCP                  6m58s
kube-system   service/kube-dns              ClusterIP   10.96.0.10       <none>        53/UDP,53/TCP,9153/TCP   6m56s
monitoring    service/alertmanager-main     ClusterIP   10.100.155.202   <none>        9093/TCP                 4m12s
monitoring    service/grafana               ClusterIP   10.98.102.194    <none>        3000/TCP                 4m11s
monitoring    service/kube-state-metrics    ClusterIP   None             <none>        8080/TCP,8081/TCP        4m11s
monitoring    service/node-exporter         ClusterIP   None             <none>        9100/TCP                 4m11s
monitoring    service/prometheus-adapter    ClusterIP   10.103.171.111   <none>        443/TCP                  4m10s
monitoring    service/prometheus-k8s        ClusterIP   10.97.193.59     <none>        9090/TCP                 4m10s
monitoring    service/prometheus-operator   ClusterIP   None             <none>        8080/TCP                 6m4s

NAMESPACE     NAME                           DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR                 AGE
kube-system   daemonset.apps/kube-proxy      1         1         1       1            1           beta.kubernetes.io/os=linux   6m56s
monitoring    daemonset.apps/node-exporter   1         1         1       1            1           kubernetes.io/os=linux        4m11s

NAMESPACE     NAME                                  READY   UP-TO-DATE   AVAILABLE   AGE
kube-system   deployment.apps/coredns               2/2     2            2           6m56s
monitoring    deployment.apps/grafana               1/1     1            1           4m11s
monitoring    deployment.apps/kube-state-metrics    1/1     1            1           4m11s
monitoring    deployment.apps/prometheus-adapter    1/1     1            1           4m10s
monitoring    deployment.apps/prometheus-operator   0/1     1            0           6m4s

NAMESPACE     NAME                                             DESIRED   CURRENT   READY   AGE
kube-system   replicaset.apps/coredns-5644d7b6d9               2         2         2       6m50s
monitoring    replicaset.apps/grafana-76b8d59b9b               1         1         1       4m11s
monitoring    replicaset.apps/kube-state-metrics-5859ffdc64    1         1         1       4m11s
monitoring    replicaset.apps/prometheus-adapter-5cd5798d96    1         1         1       4m10s
monitoring    replicaset.apps/prometheus-operator-5d94cdc9bf   1         1         0       6m4s
```
**Environment**
Mac OS, Catalina
* Prometheus Operator version:

    `Insert image tag or Git SHA here`
    
```
kubectl -n monitoring describe deployment prometheus-operator
Name:                   prometheus-operator
Namespace:              monitoring
CreationTimestamp:      Sat, 29 Feb 2020 15:39:50 -0800
Labels:                 app.kubernetes.io/component=controller
                        app.kubernetes.io/name=prometheus-operator
                        app.kubernetes.io/version=v0.36.0
Annotations:            deployment.kubernetes.io/revision: 1
Selector:               app.kubernetes.io/component=controller,app.kubernetes.io/name=prometheus-operator
Replicas:               1 desired | 1 updated | 1 total | 0 available | 1 unavailable
StrategyType:           RollingUpdate
MinReadySeconds:        0
RollingUpdateStrategy:  25% max unavailable, 25% max surge
Pod Template:
  Labels:           app.kubernetes.io/component=controller
                    app.kubernetes.io/name=prometheus-operator
                    app.kubernetes.io/version=v0.36.0
  Service Account:  prometheus-operator
  Containers:
   prometheus-operator:
    Image:      quay.io/coreos/prometheus-operator:v0.36.0
    Port:       8080/TCP
    Host Port:  0/TCP
    Args:
      --kubelet-service=kube-system/kubelet
      --logtostderr=true
      --config-reloader-image=jimmidyson/configmap-reload:v0.3.0
      --prometheus-config-reloader=quay.io/coreos/prometheus-config-reloader:v0.36.0
    Limits:
      cpu:     200m
      memory:  200Mi
    Requests:
      cpu:        100m
      memory:     100Mi
    Environment:  <none>
    Mounts:       <none>
  Volumes:        <none>
Conditions:
  Type           Status  Reason
  ----           ------  ------
  Progressing    True    NewReplicaSetAvailable
  Available      False   MinimumReplicasUnavailable
OldReplicaSets:  <none>
NewReplicaSet:   prometheus-operator-5d94cdc9bf (1/1 replicas created)
Events:
  Type    Reason             Age    From                   Message
  ----    ------             ----   ----                   -------
  Normal  ScalingReplicaSet  7m31s  deployment-controller  Scaled up replica set prometheus-operator-5d94cdc9bf to 1
```
* Kubernetes version information:

    `kubectl version`
    

* Kubernetes cluster kind:
```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.3", GitCommit:"06ad960bfd03b39c8310aaf92d1e7c12ce618213", GitTreeState:"clean", BuildDate:"2020-02-13T18:08:14Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.0", GitCommit:"2bd9643cee5b3b3a5ecbd3af49d09018f0773c77", GitTreeState:"clean", BuildDate:"2019-09-18T14:27:17Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
```
    insert how you created your cluster: kops, bootkube, tectonic-installer, etc.

* Manifests:

```
insert manifests relevant to the issue
```

* Prometheus Operator Logs:

```
Insert Prometheus Operator logs relevant to the issue here
```

* Prometheus Logs:

```
Insert Prometheus logs relevant to the issue here
```

**Anything else we need to know?**:

**Comment by brancz** [comment]
Could you share the prometheus operator logs? I just tried this myself and it works just fine. For what it's worth we can update the version documented in the readme to 1.17.3.

**Comment by russorat** [comment]
I think I'm getting something similar. Here are the logs from my prometheus-operator pod:
```

ts=2020-03-02T23:18:19.492540888Z caller=main.go:208 msg="Starting Prometheus Operator version '0.36.0'."
ts=2020-03-02T23:18:19.50143921Z caller=main.go:98 msg="Staring insecure server on :8080"
level=info ts=2020-03-02T23:18:19.511577119Z caller=operator.go:292 component=thanosoperator msg="connection established" cluster-version=v1.16.0
level=info ts=2020-03-02T23:18:19.584069452Z caller=operator.go:452 component=prometheusoperator msg="connection established" cluster-version=v1.16.0
level=info ts=2020-03-02T23:18:19.689750669Z caller=operator.go:214 component=alertmanageroperator msg="connection established" cluster-version=v1.16.0
level=info ts=2020-03-02T23:18:21.489310037Z caller=operator.go:679 component=thanosoperator msg="CRD updated" crd=ThanosRuler
level=info ts=2020-03-02T23:18:21.692499399Z caller=operator.go:653 component=alertmanageroperator msg="CRD updated" crd=Alertmanager
level=info ts=2020-03-02T23:18:21.784078089Z caller=operator.go:1920 component=prometheusoperator msg="CRD updated" crd=Prometheus
level=info ts=2020-03-02T23:18:21.804101341Z caller=operator.go:1920 component=prometheusoperator msg="CRD updated" crd=ServiceMonitor
level=info ts=2020-03-02T23:18:21.815961315Z caller=operator.go:1920 component=prometheusoperator msg="CRD updated" crd=PodMonitor
level=info ts=2020-03-02T23:18:21.82341138Z caller=operator.go:1920 component=prometheusoperator msg="CRD updated" crd=PrometheusRule
ts=2020-03-02T23:18:24.50407662Z caller=main.go:304 msg="Unhandled error received. Exiting..." err="creating CRDs failed: waiting for ThanosRuler crd failed: timed out waiting for Custom Resource: failed to list CRD: Get https://10.96.0.1:443/apis/monitoring.coreos.com/v1/prometheuses?limit=500: stream error: stream ID 31; INTERNAL_ERROR"
```

**Comment by russorat** [comment]
Upgrading to k8s 0.17.3 fixed it for me:
![image](https://user-images.githubusercontent.com/4805997/75728355-5e010480-5c9c-11ea-9fa5-4a6791c55671.png)

**Comment by kong62** [comment]
it found in kubernetes version 1.16, but ok in 1.17.3

ts=2020-03-24T03:33:52.94950328Z caller=main.go:304 msg="Unhandled error received. Exiting..." err="creating CRDs failed: waiting for ThanosRuler crd failed: timed out waiting for Custom Resource: failed to list CRD: Get https://10.96.0.1:443/apis/monitoring.coreos.com/v1/prometheuses?limit=500: stream error: stream ID 31; INTERNAL_ERROR"

**Comment by dgrisonnet** [comment]
Hi @kong62, you can fix this on `1.16` by using the `release-0.4` branch instead of master.
